node/impl/full/eth.go

```diff
@@ -1905,3 +1891,51 @@ func leftpad32(orig []byte) []byte {
 	copy(ret[needed:], orig)
 	return ret
 }
+
+func trimLeadingZeros(b []byte) []byte {
+	for i := range b {
+		if b[i] != 0 {
+			return b[i:]
+		}
+	}
+	return []byte{}
+}
+
+func cborEncodeTopicValue(orig []byte) ([]byte, error) {
+	var buf bytes.Buffer
+	err := cbg.WriteByteArray(&buf, trimLeadingZeros(orig))
+	if err != nil {
+		return nil, err
+	}
+	return buf.Bytes(), nil
+}
```

Comment:
I'm starting to doubt whether trimming leading zeroes when emitting topic values in the EVM actor is the right choice. It clearly makes logic more complicated/cumbersome in places like this, at the benefit of a minimal gas savings for the developer. I think we should back out from this decision in the upcoming FIP-0049 alignment PRs. @Stebalien -- what do you think?